ruledata.go

```diff
@@ -716,6 +688,38 @@ func SetRules(s *NetlinkConnection, content []byte) error {
 					}
 				}
 			}
+		case "file_rules":
+			vi := v.([]interface{})
+			for ruleNo := range vi {
+				rule := vi[ruleNo].(map[string]interface{})
+				path := rule["path"]
+				if path == "" {
+					log.Fatalln("Watch option needs a path!")
+				}
+				// TODO: Add key addition support
+				// key := rule["key"]
+				perms := rule["permission"]
+				log.Println("Setting File Permissions", path, perms)
+				var dd AuditRuleData
+				dd.Buf = make([]byte, 0)
+				add := AUDIT_FILTER_EXIT
+				action := AUDIT_ALWAYS
+				_audit_syscalladded = true
+
+				err := AuditSetupWatchName(&dd, path.(string))
```

Comment:
> I can't call AuditAddWatchDir here because so the AuditSetupWatchName runs the path for sanity checks and then decide on the type of the filter to be used.

of course, by calling AuditAddWatchDir I meant to place a call to AuditSetupWatchName inside  AuditAddWatchDir.

> Throughout libaudit both functions have been kept separate and are used for different purposes

ah, I suggested this after looking the libaudit.c code, I really cant find anything like audit_setup_watch_name?

The watch API  contains - 
audit_update_watch_perms 
audit_add_watch
audit_add_dir 
audit_add_watch_dir